docs/data/data-grid/accessibility/accessibility.md

```diff
@@ -17,6 +17,26 @@ Level AA exceeds the basic criteria for accessibility and is a common target for
 
 The [WAI-ARIA Authoring Practices](https://www.w3.org/WAI/ARIA/apg/patterns/grid/) provide valuable information on how to optimize the accessibility of a Data Grid.
 
+## Label
+
+You can display a label in the data grid toolbar by passing the `label` and `showToolbar` props, as shown in the demo below.
+
+{{"demo": "GridLabel.js", "bg": "inline"}}
+
+Alternatively, if there's another element in the user interface that serves as a label for the grid, you can associate it with the grid using the `aria-labelledby` prop. Set the prop to the `id` of the element that contains the descriptive text:
```

Comment:
Just realised the changes introduced here https://github.com/mui/mui-x/pull/17147 contradict the decisions made made in https://github.com/mui/mui-x/pull/15870 🤦 

Should we deprecate the `aria-label` and `aria-labelledby` props here? https://github.com/mui/mui-x/blob/master/packages/x-data-grid/src/models/props/DataGridProps.ts#L771-L778

cc @MBilalShafi

Reply:
> Should we deprecate the aria-label and aria-labelledby props here? [master/packages/x-data-grid/src/models/props/DataGridProps.ts#L771-L778](https://github.com/mui/mui-x/blob/master/packages/x-data-grid/src/models/props/DataGridProps.ts?rgh-link-date=2025-04-25T13%3A40%3A21Z#L771-L778)

They seem like the leftovers of the original PR. I'd treat it as a bugfix and remove them, considering they didn't have any affect before #17147. It'd also remove the contradiction with [this](https://mui.com/x/migration/migration-data-grid-v7/#props) breaking change.

For the users that started using meanwhile, we could offer them the alternate way (`slotProps.main`).
Wdyt?